test/src/unittests/rhythm/test_rhythmextractor2013.py

```diff
@@ -0,0 +1,192 @@
+
+    def _assertEqualResults(self, result, expected):
+        self.assertEqual(result[0], expected[0]) #bpm
+        self.assertEqualVector(result[1], expected[1]) # ticks
+        self.assertEqual(result[2], expected[2]) # confidence
+        self.assertEqualVector(result[3], expected[3]) # estimates
+        self.assertEqualVector(result[4], expected[4]) # bpmIntervals
+
+    #TODO The exact expected values for the result oof passing empty parameters is TBD
+    # (see testZero)
+    def testEmpty(self):
+        input = []
+        #expected = [0.0, [], 0.0, [], [],[]]# extra frame for confidence
+        #result = self.runInstance(input, method="multifeature")
+        #self.assertEqualVector(result, expected)
+        #result = self.runInstance(input, method="degara")
+        #self.assertEqualVector(result, expected)
+        #print(result) 
+
+    def testZero(self):
+        input = array([0.0]*10*1024) # 100 frames of size 1024
+        expected = [0.0, [], 0.0, [], [],[]] # extra frame for confidence
+        result = self.runInstance(input, method="multifeature" )
+        self._assertEqualResults(result, expected)
+        
+        expected = [1033.59375, [0.058049887,0.058049887], 0.0, [], [],[]] # extra frame for confidence
+        result = self.runInstance(input, method="degara" )
+        self._assertEqualResults(result, expected)
+
+    def testImpulseTrain(self):
+        
+        impulseTrain140 = self.pulseTrain(bpm=140, sr=44100., offset=.1, dur=10)
+        # expected values
+        expectedTicks = [i/44100. for i in range(len(impulseTrain140)) if impulseTrain140[i]!= 0]
+        expectedBpm = 139.6845550537109
+        expectedConfidence =3.785917 
+        result = self.runInstance(impulseTrain140,method="multifeature")
+        # bpm
+        self.assertAlmostEqual(result[0], expectedBpm, 1e-2)
+        # ticks
+        tickTolerance = 1.214071e+01
+        self.assertVectorWithinVector(result[1], expectedTicks, tickTolerance)
+        #confidence
+        self.assertAlmostEqual(result[2], expectedConfidence, 1e-2)
+        # estimated bpm
+        for i in range(len(result[3])):
+            self.assertAlmostEqual(result[3][i], expectedBpm, 1.)
+
+        # bpm intervals
+        for i in range(len(result[4])):
+            self.assertAlmostEqual(result[4][i], 60./expectedBpm, 0.2)
+
+        # impulse train at 90bpm no offset
+        impulseTrain90 = self.pulseTrain(bpm=90., sr=44100., offset=0., dur=20.)
+        
+        # impulse train at 200bpm with offset
+        impulseTrain200 = self.pulseTrain(bpm=200., sr=44100., offset=.2, dur=10.)
+
+        # impulse train at 200bpm with no offset
+        impulseTrain200b = self.pulseTrain(bpm=200., sr=44100., offset=0., dur=25.)
+
+        ## make a new impulse train at 140-90-200-200 bpm
+        impulseTrain = impulseTrain140 + impulseTrain90 + impulseTrain200 + \
+                       impulseTrain200b
```

Comment:
Maybe remove empty lines to group these together better.  It will be easier to separate from the previous code.